docs/data/charts/axis/axis.md

```diff
@@ -107,7 +146,13 @@ Axes rendering can be further customized. Below is an interactive demonstration
 
 {{"demo": "AxisCustomizationNoSnap.js", "hideToolbar": true, "bg": "inline"}}
 
-### Composition
+### Text Customization
+
+To customize the text elements (ticks label and the axis label) use the `tickLabelStyle` and `labelStyle` properties of the axis configuration.
+
+{{"demo": "AxisTextCustomizationNoSnap.js", "hideToolbar": true, "bg": "inline"}}
```

Comment:
WDYT about the fact that the tick labels almost overlap the tick itself with the default angle of 45 degrees and the more you rotate towards 90 degrees, the worse it gets. 🤔 
Should we maybe include some margin? Or the text management itself should account for it? 🤔 

https://github.com/mui/mui-x/assets/4941090/61d33aef-d2b8-43a6-9ae7-4c03826b37b0

Reply:
Cool, the updated demo looks great! 🙌